tests/main/interfaces-classic-content-slot/task.yaml

```diff
@@ -0,0 +1,65 @@
+summary: Ensure that the content slots for classic snaps works.
+
+details: |
+    The content-sharing interface interface allows a snap to access contents from
+    other snap. It is possible to have content slots in classic snaps, where
+    confined snaps can connect to.
+
+    A snap which defines the content sharing plug must be shown in the interfaces list.
+    The plug must be autoconnected on install and, as usual, must be able to be
+    reconnected.
+
+# We use classic snaps for the test, so it cannot run on core
+systems: [-ubuntu-core-*]
+
+environment:
+    PROVIDER_SNAP: test-snapd-classic-content-slot
+    CONSUMER_SNAP: test-snapd-content-plug-to-classic-slot
+
+restore: |
+    snap remove --purge test-snapd-content-plug-to-classic-slot
+    snap remove --purge "$PROVIDER_SNAP"
+
+execute: |
+    echo "Install content plug, check that the classic provider is not pulled"
+    snap install --edge "$CONSUMER_SNAP"
+    if snap list "$PROVIDER_SNAP"; then
+        exit 1
+    fi
+
+    echo "Install classic snap with content slot - no automatic pull for classic"
+    snap install --edge --classic "$PROVIDER_SNAP"
+
+    echo "Then the snap is listed as connected"
+    snap connections "$PROVIDER_SNAP" |
+        MATCH "$CONSUMER_SNAP:shared-content-plug \+$PROVIDER_SNAP:shared-content-slot"
+
+    echo "Remove snaps, install in reverse order"
+    snap remove --purge "$CONSUMER_SNAP"
+    snap remove --purge "$PROVIDER_SNAP"
+    snap install --edge --classic "$PROVIDER_SNAP"
+    snap install --edge "$CONSUMER_SNAP"
```

Comment:
Here, you need to use
"$TESTSTOOLS"/snaps-state install-local "$CONSUMER_SNAP"

Reply:
@sergiocazzolato thanks for the review. I think that testing the behavior when installing from the store is quite important here and it would be good to get them published there, because afaik that would be the way to check that auto-connections work. So, not sure if we want to just install locally. Does this make sense?

Reply:
It is ok to upload to the store in this case. It would be nice if snapcraft.yaml is included instead of snap.yaml to be able to build it for other architectures but amd64.